### PR TITLE
Wrap recursively, prepare release

### PR DESCRIPTION
**Pull request by jonasfj**
Single quoted and folded strings have issues encoding `foo\n bar` because space after newline is interpreted wrong.

Well, for folded strings it can be made to work, I'm not sure if we want to, or how we want to work around it -- so for now I've just ignored the bug and left a TODO.

### CODE REVIEW

**Comment by jonasfj** [comment]
Also fixes https://github.com/dart-lang/yaml_edit/issues/24

File: lib/src/strings.dart [hunk 1]
```diff
@@ -72,6 +72,15 @@ String _yamlEncodeDoubleQuoted(String string) {
 /// It is important that we ensure that [string] is free of unprintable
 /// characters by calling [assertValidScalar] before invoking this function.
```
Comment (sigurdm) [inline lib/src/strings.dart:73]: I think this comment is wrong - assertValidScalar does not check for unprintable characters.
```suggestion
/// Precondition `!_hasUnprintableCharacters(string)`.
```

File: lib/src/strings.dart [hunk 2]
```diff
@@ -72,6 +72,15 @@ String _yamlEncodeDoubleQuoted(String string) {
...
 String _tryYamlEncodeSingleQuoted(String string) {
+  // If [string] contains a newline we'll use double quoted strings instead.
+  // Single quoted strings can represent newlines, but then we have to use an
+  // empty line (replace \n with \n\n). But since leading spaces following
+  // line breaks are ignored, we can't represent "\n ".
+  // Thus, if the string contains `\n` and we're asked to do single quoted,
+  // we'll fallback to a double quoted string.
+  if (string.contains('\n')) {
```
Comment (sigurdm) [inline lib/src/strings.dart:81]: Perhaps better to update `_hasUnprintableCharacters` to also include newlines?

File: lib/src/strings.dart [hunk 3]
```diff
@@ -279,7 +294,7 @@ final Map<int, String> unprintableCharCodes = {
   8233: '\\P', //  Escaped Unicode paragraph separator (#x2029) character.
 };
 
-/// List of escape characters. In particular, \x32 is not included because it
+/// List of escape characters. In particular, \x22 is not included because it
```
Comment (sigurdm) [inline lib/src/strings.dart:297]: Was this an intentional change? - not sure I understand what this is

File: lib/src/strings.dart [hunk 4]
```diff
@@ -291,3 +306,16 @@ final Map<int, String> doubleQuoteEscapeChars = {
   47: '\\/', //  Escaped ASCII slash (#x2F), for JSON compatibility.
   92: '\\\\', //  Escaped ASCII back slash (#x5C).
 };
+
+/// List of escape characters. In particular, \x27 is not included because it
+/// can be processed normally.
+///
+/// See 5.7 Escape Characters https://yaml.org/spec/1.2/spec.html#id2776092
+final Map<int, String> singleQuoteEscapeChars = {
```
Comment (sigurdm) [inline lib/src/strings.dart:314]: Are you using this anywhere?

File: lib/src/strings.dart (hunk 1 above)
Comment (jonasfj) [reply lib/src/strings.dart:73]: Good point!

File: lib/src/strings.dart (hunk 2 above)
Comment (jonasfj) [reply lib/src/strings.dart:84]: That'll make all folded strings that contain newlines into double quoted strings...

I think this part of the code needs some refactoring. There are probably more bugs in this area.

File: lib/src/strings.dart (hunk 3 above)
Comment (jonasfj) [reply lib/src/strings.dart]: Remove the comment, because I didn't understand it.

`\x32` is supposedly "2"

**Review by sigurdm — APPROVED** [review]
LGTM

**Pull request merged**